src/plugins/discover/public/application/main/data_fetching/fetch_esql.test.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { EsHitRecord } from '@kbn/discover-utils';
+import type { ExecutionContract } from '@kbn/expressions-plugin/common';
+import { RequestAdapter } from '@kbn/inspector-plugin/common';
+import { of } from 'rxjs';
+import { dataViewWithTimefieldMock } from '../../../__mocks__/data_view_with_timefield';
+import { discoverServiceMock } from '../../../__mocks__/services';
+import { fetchEsql } from './fetch_esql';
+
+describe('fetchEsql', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('resolves with returned records', async () => {
+    const hits = [
+      { _id: '1', foo: 'bar' },
+      { _id: '2', foo: 'baz' },
+    ] as unknown as EsHitRecord[];
+    const records = hits.map((hit, i) => ({
+      id: String(i),
+      raw: hit,
+      flattened: hit,
+    }));
+    const expressionsExecuteSpy = jest.spyOn(discoverServiceMock.expressions, 'execute');
+    expressionsExecuteSpy.mockReturnValueOnce({
+      cancel: jest.fn(),
+      getData: jest.fn(() =>
+        of({
+          result: {
+            columns: ['_id', 'foo'],
+            rows: hits,
+          },
+        })
+      ),
+    } as unknown as ExecutionContract);
+    const resolveDocumentProfileSpy = jest.spyOn(
+      discoverServiceMock.profilesManager,
+      'resolveDocumentProfile'
+    );
+    expect(
+      await fetchEsql(
+        { esql: 'from *' },
+        dataViewWithTimefieldMock,
+        discoverServiceMock.data,
+        discoverServiceMock.expressions,
+        { requests: new RequestAdapter() },
+        discoverServiceMock.profilesManager
+      )
```

Comment:
> [!WARNING]  
> This function is quickly growing and becomes more difficult to maintain and use, it might be a good time for a refactor.

Reply:
Yeah I guess it's about time we clean up this function's signature. Updated here: https://github.com/elastic/kibana/pull/183797/commits/3f1ba48af0d970ebf5063c5441382abe7bf60405.